---
title: "IST2_Analysis_Montipora"
author: "Monique White"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Load Libraries
```{r, warning = FALSE, echo = FALSE, message=FALSE}
library(ggsci)
library(tidyverse)
library(dplyr)
#library(forcats)
library(reshape2)
library(stringr)
library(tidyr)
#library(tibble)
#library(sangerseqR)
#library(DECIPHER)
library(Biostrings)
library(phangorn)
library(ape)
library(ggplot2)
library(ggtree)
library(patchwork)
library(bioseq)
library(kmer)
library(GUniFrac)
library(seqinr)
library(vegan)
library(corrplot)
library(ggrepel)

library(ggmsa)
library(dendextend)
library(usedist)
```


```{r}
seqs_monti <- read_tsv("seqs.abs.monti.txt") %>% 
    filter(!(is.na(sample_name))) %>%
  dplyr::select(sample_name, `A1`:`G3ai`) %>% 
  mutate(sample_name = as.factor(sample_name)) %>% 
  filter(!grepl("-20", sample_name) & !grepl("b", sample_name))

seqs_monti
```

#Load metadata
Note the metadata requires the imput of the CoralSea2024_Metadata.xlsx. you need to change the path for where the file lives on your in individual computer.

```{r}

library(readxl)
CoralSea2024_Metadata <- read_excel("~/AIMS@JCU/CoralSea2024_Metadata.xlsx")

meta_monti <- CoralSea2024_Metadata %>% 
  rename("VialID" = "sample_name",
         "Depth (m)"= "Depth") #renamed to match the Acropora metadata that was used in ITS2 analysis

meta_monti <- meta_monti %>% 
  select(sample_name, Reference, Reef, Site,Date, Bag, Species,Depth, `Bleach score`, Vial, Year) %>% 
  filter(Species == "Montipora",
         Year != "20")
meta_monti
```

### Load Custom Functions
```{r}
read_fasta_df <- function (file = "") {
  fasta <- readLines(file)
  ind <- grep(">", fasta)
  s <- data.frame(ind = ind, from = ind + 1, to = c((ind - 
    1)[-1], length(fasta)))
  seqs <- rep(NA, length(ind))
  for (i in 1:length(ind)) {
    seqs[i] <- paste(fasta[s$from[i]:s$to[i]], collapse = "")
  }
  tib <- tibble(label = gsub(">", "", fasta[ind]), sequence = seqs)
  return(tib)
}

write_fasta_df <- function (data, filename) 
{
    fastaLines = c()
    for (rowNum in 1:nrow(data)) {
        fastaLines = c(fastaLines, as.character(paste(">", 
            data[rowNum, "label"], sep = "")))
        fastaLines = c(fastaLines, as.character(data[rowNum, 
            "sequence"]))
    }
    fileConn <- file(filename)
    writeLines(fastaLines, fileConn)
    close(fileConn)
}

dna_to_DNAbin <- function (dna){
  DNAbin <- as_DNAbin(dna)
  names(DNAbin) <- names(dna)
  return(DNAbin)
}
dna_to_DNAStringset <- function(x) 
{
    bioseq:::check_dna(x)
    DNAstr <- DNAStringSet(paste(x))
    names(DNAstr) <- names(x)
    return(DNAstr)
}

DNAStringSet_to_dna <- function(x){
    x_dna <- as_dna(paste(x))
    names(x_dna) <- names(x)
    res <- tibble(label = names(x), sequence = x_dna)
    return(res)
}

# Convert DNAstringset to DNAbin
DNAStringSet_to_DNAbin <- function(DNAStringSet){
  DNAbin <- as.DNAbin(DNAStringSet)
  return(DNAbin)
}

# https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2315-y
palette <- c("A" = "#46ff2d", 
             "G" = "#ffae01", 
             "C" = "#f24641", 
             "T" = "#4294fa", 
             "K" = "#8b4816",
             "M" = "#83831f",
             "R" = "#ffff81",
             "S" = "#ff9d80",
             "Y" = "#e381f2",
             "W" = "#80fff2",
             "V" = "#fde4b8",
             "B" = "#f9c1bf",
             "H" = "#c0d9f9",
             "D" = "#c7ffba",
             "U" = "#8989fb",
             "N" = "black", 
             "-" = "white",
             "+" = "White")


pal_df <- data.frame(names = names(palette), col = palette)
```


### Combining sequence/metadata 
```{r}
# Convert to long format
seqs_long <- seqs_monti %>%
  filter(!is.na(sample_name)) %>%
  select(sample_name, `A1`:`G3ai`) %>%
  pivot_longer(`A1`:`G3ai`) %>%
  filter(value > 0) %>% # Remove zero values
  left_join(., meta_monti)

# Q. Are we working with the post-med seqs according to the metadata in seqs?
san_check <- seqs_long %>%
  group_by(sample_name) %>%
  summarise(total = sum(value)) #A. yes

# Create a list of samples to keep that didnt fail to sequence
keepers_ss <- san_check %>%
  filter(total > 1500)

non_keep <- san_check %>% 
  filter(total < 1500)

#we filter out 2 samples with less than A142 and A191

# Filter out the failed samples
seqs_long <- seqs_long %>%
  filter(sample_name %in% keepers_ss$sample_name) %>%
  group_by(sample_name) %>%
  mutate(value_rel = value/sum(value)) %>% # Convert to relative abundance
  ungroup() %>%
  mutate(name = as.factor(name)) # Make sample names a factor

# Create a random palette for each sequence
n <- length(levels(seqs_long$name))
seqs_pal = rainbow(n, s=.6, v=.9)[sample(1:n,n, replace = FALSE)]
names(seqs_pal) <- levels(seqs_long$name)

# Read in the profile data
profiles_raw <- read_tsv("profiles.abs.monti.txt", skip = 6) %>%
    select(sample_name = `...2`, `A1-A1qd-A1qe`:`G3fd`) %>%
    filter(!is.na(sample_name)) %>%
  select(sample_name, `A1-A1qd-A1qe`:`G3fd`)

#Convert to long format 
profiles_long <- profiles_raw %>%
  pivot_longer(`A1-A1qd-A1qe`:`G3fd`) %>% # Convert it to long format
  mutate(name = paste0("p_", name)) %>% # Add a p_ to the beginning of each profile (Some profiles are single sequence profiles and clash with the Sequence names)
  filter(sample_name %in% seqs_long$sample_name) %>% # Remove samples that dont appear in the Sequence dataframe
  group_by(sample_name) %>%
  mutate(value = as.numeric(value)) %>%
  filter(value > 0) %>% # Remove 0 abundance profiles
  mutate(sample_name = as.factor(sample_name),
         name = as.factor(name)) %>% 
  ungroup() %>%
  left_join(., meta_monti) # Add in metadata

# What is the total number of profile-related sequences in each sample?
profiles_sum <- profiles_long %>%
  group_by(sample_name) %>%
  summarise(total = sum(value))

# How many sequences in each sample are not part of a profile?
residual <- left_join(profiles_sum, san_check, by = "sample_name") %>%
  mutate(residual = total.y - total.x) %>%
  select(sample_name, value = residual) %>%
  mutate(name = "non-profile sequences") %>%
  left_join(., meta_monti)

# Combine the profiles and non-profile sequences
profile_data <- rbind(profiles_long, residual) %>%
  group_by(sample_name) %>%
  mutate(value_rel = value/sum(value)) # convert to relative abundance - in that sample 

# Create palette for profiles (this is a darker palette)
n <- length(levels(profile_data$name))
profile_pal = rainbow(n, s=.6, v=.6)[sample(1:n,n, replace = FALSE)]
names(profile_pal) <- levels(profile_data$name)

# Merge the palettes and replace the non-profile sequences with grey
all_pal <- c(seqs_pal, profile_pal)
all_pal['non-profile sequences'] <- "#808080" 

# Join profiles and sequence data together into single dataframe and add more metadata
all_data <- rbind(seqs_long, profile_data) 

```

```{r}
# Convert to long format
seqs_long <- seqs_monti %>%
  filter(!is.na(sample_name)) %>%
  select(sample_name, `A1`:`G3ai`) %>%
  pivot_longer(`A1`:`G3ai`) %>%
  filter(value > 0) %>% # Remove zero values
  left_join(., meta_monti)

# Q. Are we working with the post-med seqs according to the metadata in seqs?
san_check <- seqs_long %>%
  group_by(sample_name) %>%
  summarise(total = sum(value)) #A. yes

# Create a list of samples to keep that didnt fail to sequence
keepers_ss <- san_check %>%
  filter(total > 1500)

non_keep <- san_check %>% 
  filter(total < 1500)

#we filter out 2 samples with less than A142 and A191

# Filter out the failed samples
seqs_long <- seqs_long %>%
  filter(sample_name %in% keepers_ss$sample_name) %>%
  group_by(sample_name) %>%
  mutate(value_rel = value/sum(value)) %>% # Convert to relative abundance
  ungroup() %>%
  mutate(name = as.factor(name)) # Make sample names a factor

# Create a random palette for each sequence
n <- length(levels(seqs_long$name))
seqs_pal = rainbow(n, s=.6, v=.9)[sample(1:n,n, replace = FALSE)]
names(seqs_pal) <- levels(seqs_long$name)

# Read in the profile data
profiles_raw <- read_tsv("profiles.abs.monti.txt", skip = 6) %>%
    select(sample_name = `...2`, `A1-A1qd-A1qe`:`G3fd`) %>%
    filter(!is.na(sample_name)) %>%
  select(sample_name, `A1-A1qd-A1qe`:`G3fd`)

#Convert to long format 
profiles_long <- profiles_raw %>%
  pivot_longer(`A1-A1qd-A1qe`:`G3fd`) %>% # Convert it to long format
  mutate(name = paste0("p_", name)) %>% # Add a p_ to the beginning of each profile (Some profiles are single sequence profiles and clash with the Sequence names)
  filter(sample_name %in% seqs_long$sample_name) %>% # Remove samples that dont appear in the Sequence dataframe
  group_by(sample_name) %>%
  mutate(value = as.numeric(value)) %>%
  filter(value > 0) %>% # Remove 0 abundance profiles
  mutate(sample_name = as.factor(sample_name),
         name = as.factor(name)) %>% 
  ungroup() %>%
  left_join(., meta_monti) # Add in metadata

# What is the total number of profile-related sequences in each sample?
profiles_sum <- profiles_long %>%
  group_by(sample_name) %>%
  summarise(total = sum(value))

# How many sequences in each sample are not part of a profile?
residual <- left_join(profiles_sum, san_check, by = "sample_name") %>%
  mutate(residual = total.y - total.x) %>%
  select(sample_name, value = residual) %>%
  mutate(name = "non-profile sequences") %>%
  left_join(., meta_monti)

# Combine the profiles and non-profile sequences
profile_data <- rbind(profiles_long, residual) %>%
  group_by(sample_name) %>%
  mutate(value_rel = value/sum(value)) # convert to relative abundance - in that sample 

# Create palette for profiles (this is a darker palette)
n <- length(levels(profile_data$name))
profile_pal = rainbow(n, s=.6, v=.6)[sample(1:n,n, replace = FALSE)]
names(profile_pal) <- levels(profile_data$name)

# Merge the palettes and replace the non-profile sequences with grey
all_pal <- c(seqs_pal, profile_pal)
all_pal['non-profile sequences'] <- "#808080" 

# Join profiles and sequence data together into single dataframe and add more metadata
all_data <- rbind(seqs_long, profile_data) 
```

```{r}
# How many samples per species?
all_data %>%
  distinct(sample_name, Reef) %>%
  group_by(Reef) %>% 
  summarise(total_samples = n())

study_total <- all_data %>% 
   filter(!(str_detect(name, "p_")),
         name != "non-profile sequences") %>% 
  summarise(total_seqs = sum(value)) %>%
  pull(total_seqs)

#11438183 sequences in Montipora dataset
```


#By symbiont genus
Please note I was a bit confused by some of the calculations here, so I have hashed them out.
```{r}
all_data %>%
  filter(!(str_detect(name, "p_")), name != "non-profile sequences") %>%
  filter(str_sub(name, 1, 1) == "C" | str_detect(name, "_C")) %>% 
  summarise(sum = sum(value))
# 11090862	

## not sure this line(X / study_total) * 100 # X Cladocopium proportion of all data.

#Total number of Symbiodinium
all_data %>%
  filter(!(str_detect(name, "p_")), name != "non-profile sequences") %>%
  filter(str_sub(name, 1, 1) == "A" | str_detect(name, "_A")) %>% 
  summarise(sum = sum(value)) 
# 256050		total seqs
## Not sure about this calculation (X	 / study_total) * 100 # X Symbiodium


# Total number of Durusdinium sequences
all_data %>%
  filter(!(str_detect(name, "p_")), name != "non-profile sequences") %>%
  filter(str_sub(name, 1, 1) == "D" | str_detect(name, "_D")) %>% 
  summarise(sum = sum(value)) 
# 76907 total reads
```


```{r}
library(tidyverse)

#total type profiles across Montipora 
all_data %>%
  filter(str_detect(name, "p_")) %>%    #profiles start with p_
  group_by(name) %>%
  dplyr:: count() %>%
  dplyr:: arrange(desc(n)) %>%
  ungroup() %>%
  mutate(prop = n/sum(n)) %>%
  mutate(cumulative_sum = cumsum(prop)) %>% 
  print(n=73) #to get a list of the type profiles in montipora

all_data %>%
  filter(str_detect(name, "p_")) %>%    #profiles start with p_
  group_by(name) %>%
  dplyr:: count() %>%
  dplyr:: arrange(desc(n)) %>%
  ungroup() %>%
  mutate(prop = n/sum(n)) %>%
  mutate(cumulative_sum = cumsum(prop)) %>% 
  print(n=73) %>%  #to get a list of the type profiles in montipora
write.csv(file = "Montipora_ITS2_Type_Profiles.csv")
```


Not sure this line works, not quite sure what I am meant to be doing here though.

```{r eval = FALSE}
all_data %>% 
  filter(str_detect(name, "p_")) %>%
  group_by(sample_name) %>% 
  summarise(n = n()) %>% 
 #filter(n == 1)     #147 samples have 1 type profile
 #filter(n == 2)     #34 samples have 2 type profiles 
#filter(n == 3)     #3 samples have 3 profiles 
```

```{r}
all_data %>% 
    filter(str_detect(name, c("p_"))) %>% 
    filter(str_detect(name, c("p_C3k|p_C3"))) %>% 
  summarise(mean = mean(value_rel))      #for relative proportion

#Either C3k or C3 were found as dominant seq in 63% of corals sampled 
```


```{r}
#filter out non-profile sequences
seq_data <- all_data %>% 
  filter(!str_detect(name, "non")) %>% 
  filter(!str_detect(name, "p_"))
```

#2.1 UPGMA tree
Not sure when seq_data was made in the Acropora analysis?
```{r}
fasta_monti <- read_fasta_df("seqs.monti.fasta") %>% 
   filter(label %in% seqs_long$name) %>%   #only keeping DNA seqs that appear in seqs_long subset 
  deframe() %>%
  as_dna()
```

```{r eval =FALSE}
#creating the tree
kdist_monti <- fasta_monti %>%
  dna_to_DNAbin() %>%
  kdistance(k = 7, residues = "DNA", method = "edgar") %>%
  as.matrix()
```

```{r eval=FALSE}
k_tree_monti <- kdist_monti %>% phangorn::upgma()

k_tree_monti
```

```{r}
seqs_wide_monti <- seq_data %>%
  dplyr::select(sample_name, name, value) %>%
  pivot_wider(names_from = name, values_from = value, values_fill = 0) %>%
  tibble::column_to_rownames(var = "sample_name")

head(seqs_wide_monti)
```

```{r}
#k_unidist_monti <- GUniFrac(seqs_wide_monti, k_tree_monti)   #GUniFrac calculates all the distances 
#k_unidist_monti <- k_unidist_monti$unifracs

load("~/AIMS@JCU/CoralSeaSymbioants/ITS2 Montipora/Montipora_CS_Analysis/k_unidist_monti.RData")

 # GUniFrac with alpha 0.5 (i.e Generalised UniFrac Distance)
du_monti <- k_unidist_monti[, , "d_0.5"]   
dist_monti <- as.dist(du_monti, diag = FALSE)

# Cluster the samples
hclust_samps_monti <- upgma(du_monti)

# Make the sample tree
tree_monti <- ggtree(hclust_samps_monti, size = 0.2) +
  theme(aspect.ratio = 0.3) + layout_dendrogram()

# Get a sample order from ggtree
monti_sample_order <- tree_monti$data %>% filter(isTip == "TRUE") %>%
  arrange(y) %>%
  pull(label)

# Start plotting the composition data
plot_df_monti <- all_data %>%
  mutate(sample_name = fct_relevel(sample_name, monti_sample_order))

theme_set(theme_bw())


# find the likely distinguishing seqs in here
test_df <- seq_data %>%
   group_by(name) %>%
   summarise(mean = mean(value_rel), n = n()) %>%
   arrange(desc(n), desc(mean))

# colour them black to check
 test_pal <- all_pal
# test_pal['C21ab'] <- "#000000" 
```

## Fig 2B
```{r}
bar_uni_monti <- 
ggplot(plot_df_monti, aes(sample_name, value_rel)) +
geom_bar(stat = "identity", aes(fill = name, colour = name)) +
theme(aspect.ratio = 0.5, legend.position = "none", axis.text.y=element_blank(), axis.ticks.y = element_blank(),
      axis.text.x=element_blank(), axis.ticks.x = element_blank(),
      axis.title.x = element_blank(), axis.title.y = element_blank(),
      panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.ticks = element_blank()) +
scale_fill_manual(values = all_pal, breaks = levels(profile_data$name)) +
scale_colour_manual(values = all_pal, breaks = levels(profile_data$name)) +
geom_hline(yintercept = 1, size = 1) +
guides(fill=guide_legend(ncol=2))


#p_bar_uni is the sequences by colour. P_tree_tip is the tree coloured by reef. 

tree_monti / bar_uni_monti
```


```{r}
# Create reef strip
reef_strip <- ggplot(plot_df_monti, aes(x = sample_name, y = 1, fill = Reef)) +
  geom_tile() +
  theme_void() +
  theme(legend.position = "top") #+
 # scale_fill_brewer(palette = "Set2")  # or use your own palette

# Now combine the reef strip and your barplot
combined_plot <- reef_strip / bar_uni_monti + plot_layout(heights = c(0.1, 1))

# Display it
combined_plot
```


###4.1.3 monti Reef PCoA
```{r}
pcoa_monti <- cmdscale(dist_monti, eig = TRUE)    #this is doing 'cmds' or classic multidimensional scaling

#outlier_samples <- X

ordiplot(pcoa_monti, display = 'sites', type = 'text')
barplot (pcoa_monti$eig, names = paste ('PCoA', 1:X), las = X, ylab = 'eigenvalues')

# Custom x and y labels
xlab = paste("Axis 1 (X)")
ylab = paste("Axis 2 (X)")

MDSxy.monti <- data.frame(pcoa_monti$points) %>% 
  rownames_to_column(var = "sample_name") %>% 
  left_join(., meta_monti)# %>% 
 #  mutate(catBleaching = as.factor(catBleaching)) %>% 
 # filter(!(Vial %in% outlier_samples)) 

montiPCA <- MDSxy.monti %>%
  filter(Reef != "NA") %>% 
 # mutate(Reef = fct_relevel(Reef, reef_order)) %>%
  ggplot(aes(X1, X2, fill = Reef)) + 
  geom_point(alpha = 1, shape = 21, size = 3) + 
  scale_fill_viridis_d(option = "magma", direction = -1)+ 
#  scale_fill_manual(values = reef_pal) #+
 theme(legend.position = "none", aspect.ratio = 1, text = element_text(size = 15))

montiPCA
```

## Procrustes Rotation

```{r}
load("~/AIMS@JCU/CoralSeaSymbioants/ITS2 Montipora/Montipora_CS_Analysis/ind_coords_monti.RData")

monti_seqs_host <- ind_coords_monti %>% 
  mutate("sample_name" = VialID)

#code for defining specific colours per reef for procrustes plots later on
reef_pal2 <- c("#ac1917", "#e54a50", "#b75420", "#ea8553", "#c0982b", "#eec76b", "#768b45", "#a9be77", "#B2DFDB", "#4eaaa6", "#00798C")

reef_names <- c("Elizabeth", "Middleton", "Cato", "Wreck", "Kenn", "Mellish", "Marion","Frederick", "Saumarez", "Madgelaine", "NA")

names(reef_pal2) <- reef_names
```

```{r}
#here is the subset with the data that matches the host data
shared_seqs_monti <- inner_join(monti_seqs_host, seq_data, by = "sample_name")

seq_data %>% 
  distinct(sample_name) %>% 
  nrow()
 #should be 187 in the symbioant

shared_seqs_monti <- shared_seqs_monti %>% 
  select(sample_name)
```

Host tree

```{r}
 load("~/AIMS@JCU/CoralSeaSymbioants/ITS2 Montipora/Montipora_CS_Analysis/monti_gen.RData") #load host data

library(adegenet) #install required package
library(poppr)


# Check your individual names in monti_gen
ind_names_host <- indNames(monti_gen)

# Check the sample names you want to keep
sample_names <- shared_seqs_monti$sample_name

# Subset the genind object to retain only matching individuals
monti_gen_subset <- monti_gen[ind_names_host %in% shared_seqs_monti$sample_name, ]

length(indNames(monti_gen))          # Before 180
length(indNames(monti_gen_subset))   # After 177

#Compute the pairwise distances between multivariate observations; used on raw allele frequencies.
host_shared_dist <- dist(monti_gen_subset, method = "euclidean") #calculate euclidean distance 

# convert the distance matrix to a dist object
host_dist_matrix <- as.dist(host_shared_dist)

#Create a matrix with only common sample names for symbioant
seqs_wide_symbioant <- seq_data %>%
  select(sample_name, name, value) %>%
  filter(sample_name %in% shared_seqs_monti$sample_name) %>% 
  pivot_wider(names_from = name, values_from = value, values_fill = 0) %>%
  tibble::column_to_rownames(var = "sample_name") #187 samples

#k_tree_monti_symbioant <- kdist_monti_symbioant %>% phangorn::upgma()

load("~/AIMS@JCU/CoralSeaSymbioants/ITS2 Montipora/Montipora_CS_Analysis/k_unidist_monti.RData")
k_unidist_symbioant <- k_unidist_monti


 # GUniFrac with alpha 0.5 (i.e Generalised UniFrac Distance)
du_monti_symbioant <- k_unidist_symbioant[, , "d_0.5"]   
dist_monti_symbioant <- as.dist(du_monti_symbioant, diag = FALSE)
```

# Symbioant Tree 



```{r}
#convert host (H) and symbiont (P) distance trees as data matrices
H.dist <- host_dist_matrix %>% as.matrix()
P.dist <- dist_monti_symbioant %>% as.matrix() 
```

```{r}
#must have same number of dimensions 
dim(H.dist)
dim(P.dist) 

# Subset to only the samples that are in both datasets
intersect_samples <- intersect(rownames(H.dist), rownames(P.dist))
intersect_samples_col <- intersect(colnames(H.dist), colnames(P.dist)) # The host data set had 177, and the symbioant had 187
P.dist_intersect <- P.dist[intersect_samples, intersect_samples_col]
H.dist_intersect <- H.dist[intersect_samples, intersect_samples_col ]

# Subset to only the samples that are in both datasets
intersect_samples <- intersect(rownames(H.dist_intersect), rownames(P.dist_intersect))
P.dist <- P.dist_intersect
H.dist <- H.dist_intersect

#recheck dimensions - all good 
dim(P.dist)
dim(H.dist)
```

### Perform Paco analysis

```{r, include = FALSE}
# Perform Procrustes analysis. logical that symbionts depend on the host. so X = host and Y = symbiont. 

paco.monti <- procrustes(H.dist, P.dist, symmetric = FALSE)
summary(paco.monti)

#symmetric = FALSE: This specifies that the Procrustes rotation can be non-orthogonal (i.e. generalised), meaning that it can include scaling and shearing transformations in addition to rotation and reflection. This allows for more flexible alignment between the two dissimilarity matrices, but may result in a less interpretable or meaningful alignment depending on the nature of the data.


#statistics check for residuals 
#paco error chart 
plot(paco.monti, kind = 1, type = "text")

#show paco residuals. higher numbers mean these are the samples that have high dissimilarity between the symbiont and host trees
plot(paco.monti, kind = 2, type = "text")
```

Call:
procrustes(X = H.dist, Y = P.dist, symmetric = FALSE) 

Number of objects: 176    Number of dimensions: 176 

Procrustes sum of squares:  
  3385946 
Procrustes root mean squared error: 
 138.7023 
Quantiles of Procrustes errors:
      Min        1Q    Median        3Q       Max 
 74.55555 105.16398 126.53689 155.95953 352.93210


### montipora procrustes plot

```{r}
#transfer PC coords into dataframe for plotting  
hp.df.monti <- data.frame(rda1=paco.monti$Yrot[,1],
rda2=paco.monti$Yrot[,2],xrda1=paco.monti$X[,1],
xrda2=paco.monti$X[,2]) %>% 
  tibble::rownames_to_column(var = "Vial") 


#filter metadata to just reef
meta.reef <- meta_monti %>% 
  select(sample_name, Reef) %>% 
  rename(sample_name = "Vial")

#left join the metadata to get reef into the dataframe 
hp.df.monti2 <- hp.df.monti %>% left_join(meta.reef)

#create plot 
procrust.monti <- ggplot(hp.df.monti2) +
  geom_segment(aes(x=rda1,y=rda2,xend=xrda1,yend=xrda2), size = .2) +
  geom_point(aes(x=rda1, y=rda2, shape = "Host", color = Reef)) +
  geom_point(aes(x=xrda1, y=xrda2, shape = "Symbiont", color = Reef)) +
  scale_shape_manual(name = "Groups", 
                     values = c("Host" = 19, "Symbiont" = 17)) +
  scale_colour_manual(values = reef_pal2) +
    theme(legend.position = "none") +
  theme(plot.background = element_blank(),
    panel.grid.minor = element_blank(),
    panel.grid.major = element_blank()) +
  theme(aspect.ratio = 1) 

save(procrust.monti, file = "procrust.monti.RData")
```

### Test significance 

```{r}
#testing significance 
protest(X = H.dist, Y = P.dist, scores = "sites", permutations = 999)
```

